lib/gollum/app.rb

```diff
@@ -179,7 +179,7 @@ def wiki_new
       halt 500 unless tempfile.is_a? Tempfile
 
       # Remove page file dir prefix from upload path if necessary -- committer handles this itself
-      dir      = wiki.per_page_uploads ? params[:upload_dest].match(/^(#{wiki.page_file_dir}\/+)?(.*)/)[2] : 'uploads'
+      dir      = wiki.per_page_uploads ? params[:upload_dest] : ::File.join([wiki.page_file_dir, 'uploads'].compact)
```

Comment:
> Handling `page-file-dir` only internally will actually make this much easier, since we can then be sure that any request paths from the client don't contain the `page-file-dir` prefix yet.

Yep.

> So there should just be a helper method on the `Wiki` class that converts request paths to in-repo paths (and one to do the opposite I guess).

Almost what I was suggesting :).

> This being ruby I think there's no way to 'enforce' the use of a particular method to construct a path, nor should there be.

Using only a single method everywhere is not an option of course, there are some "variables". But why couldn't we come up with several methods that do the hardest "internal" work, common to all cases?

> For now we have a serious bug, and we should just squash it instead of waiting for someone to do the Right Thing Soon(tm).

Sure, nothing against fixing a bug or two. I was just trying to get your attention to the big (imo) issue :).